pywinauto/controls/uia_controls.py

```diff
@@ -926,9 +928,88 @@ def texts(self):
+        first_pos_y = btn_y_coord
+        elem_y_pos = btn_y_coord
+        menu_elem_list = []
+        direction = 0
+
+        while direction <= 1:
+            button_elem_info = UIAElementInfo.from_point(btn_x_coord, elem_y_pos)
+
+            if not direction and button_elem_info.control_type == 'MenuItem':
+                elem_height = button_elem_info.rectangle.height()
+                elem_width = button_elem_info.rectangle.width()
+                right_x = btn_x_coord + elem_width
+                left_x = btn_x_coord - elem_width
+                menu_elem_list.append(button_elem_info)
+
+                if self._menu_item_exists(right_x, elem_y_pos) and side == 'right':
+                    # Check right side from the item
+                    menu_elem_list += self._up_down_menu_items(right_x, elem_y_pos, side='right')
+                elif self._menu_item_exists(left_x, elem_y_pos):
+                    # Check left side from the item
+                    menu_elem_list += self._up_down_menu_items(left_x, elem_y_pos, side='left')
+
+                elem_y_pos += elem_height
+
+            elif not direction and button_elem_info.control_type != 'MenuItem':
+                direction = 1
+                elem_y_pos = first_pos_y
+
+            if direction and button_elem_info.control_type == 'MenuItem':
+                elem_height = button_elem_info.rectangle.height()
+                elem_width = button_elem_info.rectangle.width()
+                right_x = btn_x_coord + elem_width
+                left_x = btn_x_coord - elem_width
+                menu_elem_list.append(button_elem_info)
+
+                if self._menu_item_exists(right_x, elem_y_pos) and side == 'right':
+                    # Check right side from the item
+                    menu_elem_list += self._up_down_menu_items(right_x, elem_y_pos, side='right')
+                elif self._menu_item_exists(left_x, elem_y_pos):
+                    # Check left side from the item
+                    menu_elem_list += self._up_down_menu_items(left_x, elem_y_pos, side='left')
+
+                elem_y_pos -= elem_height
+
+            elif direction and button_elem_info.control_type != 'MenuItem':
+                direction = 2
+
+        return menu_elem_list
+
+    def _collect_menu_items(self, btn_count):
+        # Collect element info from toolbar buttons
+        buttons_elem_list = []
+        cc = []
+        for btn_num in range(btn_count):
+            button_coord = common_controls.ToolbarWrapper(self.element_info.handle) \
+                .get_button_rect(btn_num).mid_point()
+            button_coord_x, button_coord_y = self.client_to_screen(button_coord)
+
+            buttons_elem_list += self._up_down_menu_items(button_coord_x, button_coord_y)
+
+            for btn_elem_info in buttons_elem_list:
+                ctrl_text = btn_elem_info.name
+                added_ctrl_text = [el.window_text() for el in cc]
```

Comment:
I would suggest creating `cc_texts = set()` in addition to `cc` so that we don't walk `cc` many times. Also checks like `if ctrl_text not in cc_texts` will be faster because `set` uses hash function to calculate the element address in the data structure: it doesn't require walking the whole list.